lib/ansible/modules/package_facts.py

```diff
@@ -552,21 +552,18 @@ def main():
 
         seen.add(pkgmgr)
         try:
-            try:
-                # manager throws exception on init (calls self.test) if not usable.
-                manager = PKG_MANAGERS[pkgmgr]()
-                if manager.is_available():
-                    found += 1
+            # manager throws exception on init (calls self.test) if not usable.
+            manager = PKG_MANAGERS[pkgmgr]()
+            if manager.is_available():
+                found += 1
+                try:
                     packages.update(manager.get_packages())
-
-            except Exception as e:
-                if pkgmgr in module.params['manager']:
-                    module.warn('Requested package manager %s was not usable by this module: %s' % (pkgmgr, to_text(e)))
-                continue
+                except Exception as e:
+                    module.warn('Failed to retrieve packages with %s: %s' % (pkgmgr, to_text(e)))
 
         except Exception as e:
             if pkgmgr in module.params['manager']:
-                module.warn('Failed to retrieve packages with %s: %s' % (pkgmgr, to_text(e)))
+                module.warn('Requested package manager %s was not usable by this module: %s' % (pkgmgr, to_text(e)))
```

Comment:
@flowerysong Could you please add a test for this change?

Reply:
rest LGTM

Reply:
...but if I do that I might find out that things are broken and have to fix them. Bleh.

(The manager does not throw exceptions on init, nor indeed on `is_available()`.)

Reply:
At least we will cover the regression here.

Reply:
> ...but if I do that I might find out that things are broken and have to fix them.

If you don't, where would we have a proof that your patch …actually works?